web-common/src/features/dashboards/time-dimension-details/charts/tdd-tooltip-formatter.ts

```diff
@@ -41,21 +50,36 @@ export function tddTooltipFormatter(
       content += "<table>";
 
       const selectedValuesLength = selectedDimensionValues.length;
-      if (chartType === TDDChart.STACKED_AREA && selectedValuesLength) {
+      if (
+        chartType === TDDChart.STACKED_AREA &&
+        (selectedValuesLength || isTimeComparison)
+      ) {
         for (const key of keys) {
           const val = rest[key];
-
           if (val === null || val === "NaN") continue;
 
+          let label = key;
+          let keyColor = colorMap[String(key)] || "#000";
+
+          if (isTimeComparison) {
+            if (key.startsWith("comparison.")) {
+              keyColor = colorMap["comparison_period"];
+              label = "Previous Period";
+            } else {
+              keyColor = colorMap["current_period"];
+              label = "Current Period";
+            }
+          }
```

Comment:
**Performance Issue**: The expanded conditional logic and repeated checks within the tooltip formatter can significantly impact performance, especially with complex conditional structures and large data. <br> **Fix**: Simplify and optimize the conditional logic in the tooltip formatter to reduce complexity and improve performance. <br> **Code Suggestion**: 
 ```
 +          const { label, keyColor } = getLabelAndColor(key, isTimeComparison);  // This function abstracts away the complex logic
 +          content += formatTableRow(label, keyColor, val);  // Use a function to format the table row HTML
 ```

Reply:
Agree that this function looks fairly complex due to the nested conditionals. Maybe either guard clauses, more comments, and/or helper functions could help make it more readable.